src/gui/Application.cpp

```diff
@@ -326,7 +341,17 @@ bool Application::isAlreadyRunning() const
     return config()->get(Config::SingleInstance).toBool() && m_alreadyRunning;
 }
 
-bool Application::sendFileNamesToRunningInstance(const QStringList& fileNames)
+/**
+ * Send data to the running UI instance; currently used to lock databases via the CLI or send new files to open.
+ *
+ * Right now nullptr fileNames is sent as long as it is not empty. This is not sustainable; the sent data (if any)
+ * should be as flexible enough to enable future changes as much room to breathe as possible. For now, though, I just
+ * want both file names and locking to work, and for that, this is sufficient. --wundrweapon
+ * @param id - command ID to send. Right now, 1 opens files and 2 locks databases (ignoring the fileNames param)
+ * @param fileNames - list of file names to open
+ * @return true if all operations succeeded (connection made, data sent, connection closed)
+ */
+bool Application::sendToRunningInstance(quint32 id, const QStringList& fileNames)
```

Comment:
You are abusing this function to do two completely different things depending on what is passed to the parameters. You should instead split the function:
`Application::sendFilenamesToInstance(filenames)`
`Application::sendLockToInstance()`